src/dbg/debugger.cpp

```diff
@@ -1921,39 +1927,52 @@ bool dbglistprocesses(std::vector<PROCESSENTRY32>* infoList, std::vector<std::st
         char* cmdline;
 
         if(!dbggetcmdline(&cmdline, NULL, hProcess))
-            commandList->push_back("");
+            commandList->push_back("ARG_GET_ERROR");
         else
         {
             cmdline_qoutes_placement_t posEnum = getqoutesplacement(cmdline);
             char* cmdLineExe = strstr(cmdline, pe32.szExeFile);
-            duint cmdLineExeSize = cmdLineExe ? strlen(pe32.szExeFile) : 0;
+            size_t cmdLineExeSize = cmdLineExe ? strlen(pe32.szExeFile) : 0;
+
+            if(!cmdLineExe)
+            {
+                char* exeName = strrchr(pe32.szExeFile, '\\') ? strrchr(pe32.szExeFile, '\\') + 1 :
+                                strrchr(pe32.szExeFile, '/') ? strrchr(pe32.szExeFile, '/') + 1 : pe32.szExeFile;
+
+                char* exeNameInCmd = strstr(cmdline, exeName);
+                if(exeNameInCmd)
+                    cmdLineExeSize = (size_t)(((LPBYTE)exeNameInCmd - (LPBYTE)cmdline) + strlen(exeName));
+                else
+                {
+                    //try to locate basic name, without extension
+                    char* basicName = (char*)emalloc(strlen(exeName) + 1, "dbglistprocesses:basicName");;

```

Comment:
This should be `Memory<char*> basicName(strlen(exeName) + 1);` which ensures the buffer is always freed on scope exit. Doesn't matter much though I can change it during the merge.